schemas/com.snowplowanalytics.badrows/enrichment_failures/jsonschema/1-0-0

```diff
@@ -0,0 +1,115 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for a bad row resulting from enrichment failures",
+  "self": {
+    "vendor": "com.snowplowanalytics.badrows",
+    "name": "enrichment_failures",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "failure": {
+      "type": "object",
+      "description": "Information regarding the enrichment failures",
+      "properties": {
+        "timestamp": {
+          "type": "string",
+          "description": "Timestamp at which the failure occurred",
+          "format": "date-time"
+        },
+        "messages": {
+          "type": "array",
+          "description": "List of failure messages associated with the enrichment failures",
+          "items": {
+            "type": "object",
+            "description": "Individual enrichment failure",
+            "anyOf": [
+              {
+                "properties": {
+                  "enrichment": {
+                    "type": "string",
+                    "description": "Enrichment which produced this failure",
+                    "pattern": "^iglu:[a-zA-Z0-9-_.]+/[a-zA-Z0-9-_]+/[a-zA-Z0-9-_]+/[0-9]+-[0-9]+-[0-9]+$"
+                  },
+                  "message": {
+                    "type": "string",
+                    "description": "Failure message associated with this enrichment failure",
+                    "maxLength": 256
+                  }
+                },
+                "required": [ "enrichment", "message" ],
+                "additionalProperties": false
+              },
+              {
+                "properties": {
+                  "etc": {
+                    "type": "string"
+                  }
+                },
+                "required": [ "etc" ],
+                "additionalProperties": false
+              }
+            ]
+          }
+        }
+      }
+    },
+    "payload": {
```

Comment:
As discussed with Anton, there are two options for payloads we incorporate in bad rows happening during schema validation and enrichment:

- HTTP payload, this might help during exploring since the payload schema is consistent for all bad row types (except storage)
- tracker protocol payload + collector metadata, this is better structured and reflect that Snowplow has "understood" the incoming payload, the burden then falls to the recovery process to reconstruct a collector payload from this data

Reply:
As discussed with @BenFradet:

For failures at the first three stages of the pipeline (up to Tracker Protocol Failure) the payload will be the raw HTTP payload.

For schema validation and enrichment errors the payload will be the tracker protocol + collector metadata.

For load failures the payload will be the enriched event.